openhands/cli/tui.py

```diff
@@ -700,20 +700,207 @@ def _(event: KeyPressEvent) -> None:
+    if command.strip().startswith('|') or command.strip().endswith('|'):
+        return []  # Invalid: starts or ends with pipe
+
+    if '||' in command:
+        return []  # Invalid: double pipes
+
+    # Split by pipe operator, handling quoted strings properly
+    parts = []
+    current_part: list[str] = []
+
+    # Use shlex to properly handle quoted strings and escapes
+    try:
+        tokens = shlex.split(command, posix=True)
+    except ValueError:
+        # If shlex fails (e.g., unmatched quotes), fall back to simple split
+        tokens = command.split()
+
+    # Reconstruct the command while splitting on pipes
+    i = 0
+    while i < len(tokens):
+        token = tokens[i]
+        if token == '|':
+            if current_part:
+                parts.append(' '.join(current_part))
+                current_part = []
+            else:
+                # Empty part before pipe - invalid
+                return []
+        else:
+            current_part.append(token)
+        i += 1
+
+    # Add the last part if it exists
+    if current_part:
+        parts.append(' '.join(current_part))
+    elif parts:  # We had pipes but no final command
+        return []  # Invalid: ends with pipe
+
+    # If no pipes were found, fall back to simple pipe splitting
+    if len(parts) <= 1 and '|' in command:
+        simple_parts = [part.strip() for part in command.split('|')]
+        # Check for empty parts
+        if any(not part for part in simple_parts):
+            return []  # Invalid: empty parts
+        parts = simple_parts
+
+    return parts
+
+
+def _generate_command_pattern(command: str) -> str:
+    """Generate a regex pattern for a command that can match similar commands.
+
+    This function handles both simple commands and piped commands.
+    For piped commands, it generates patterns for each sub-command.
+
+    Args:
+        command: The command to generate a pattern for (may contain pipes).
+
+    Returns:
+        str: A regex pattern that matches similar commands.
+    """
+    # Parse the command to handle pipes
+    sub_commands = _parse_piped_command(command)
+
+    if len(sub_commands) == 1:
+        # Single command, use the original logic
+        return _generate_single_command_pattern(command)
+    else:
+        # Piped command, generate pattern for each sub-command
+        sub_patterns = []
+        for sub_command in sub_commands:
+            sub_pattern = _generate_single_command_pattern(sub_command.strip())
+            # Remove the ^ and $ anchors from sub-patterns
+            sub_pattern = (
+                sub_pattern[1:-1]
+                if sub_pattern.startswith('^') and sub_pattern.endswith('$')
+                else sub_pattern
+            )
+            sub_patterns.append(sub_pattern)
+
+        # Join sub-patterns with pipe separator (allowing flexible whitespace around pipes)
+        pattern = '\\s*\\|\\s*'.join(sub_patterns)
+        return f'^{pattern}$'
```

Comment:
@Openhands can we instead return multiple sub patterns and ask the user one by one which one they want to accept?

We should also update openhands/cli/tui.py file so that user can actually edit the pattern if they want (e.g., if they want to make it more specific, instead of `git`, they want to only match and allow `git add`. -- make sure you print these as instruction for the user).